src/databricks/labs/lsql/dashboards.py

```diff
@@ -120,10 +124,15 @@ def _get_widget(self, dataset: Dataset) -> Widget:
         query = Query(dataset_name=dataset.name, fields=fields, disaggregated=True)
         # As far as testing went, a NamedQuery should always have "main_query" as name
         named_query = NamedQuery(name="main_query", query=query)
-        # Counters are expected to have one field
-        counter_field_encoding = CounterFieldEncoding(field_name=fields[0].name, display_name=fields[0].name)
-        counter_spec = CounterSpec(CounterEncodingMap(value=counter_field_encoding))
-        widget = Widget(name=dataset.name, queries=[named_query], spec=counter_spec)
+        spec: WidgetSpec
+        if len(fields) == 1:  # Counters are expected to have one field
+            counter_encodings = CounterFieldEncoding(field_name=fields[0].name, display_name=fields[0].name)
```

Comment:
can you refactor this code to introduce three new classes: `CounterTile`, `TableTile`, and `MarkdownTile`. Those classes should have a `widget_spec() -> WidgetSpec` and `size() -> tuple[int, int]` methods.

Reply:
Sure, do you want each of them to receive the fields on initialisation? And the `WidgetMetdata`, how should that be merged with the size?